src/providers/analytics/analytics.ts

```diff
@@ -1,44 +1,134 @@
+        })
+        .catch((error) => console.log(`enableExceptionReporting: ${this.analyticsStartupError}`, error));
+    });
   }
 
   setCurrentPage(name: string) {
-    this.platform.ready().then(() => {
+    this.platform.ready().then(() => { 
       this.ga
         .startTrackerWithId(this.googleAnalyticsKey)
         .then(() => {
-          this.ga.trackView(name).then((resp) => {});
+          this.ga.trackView(name)
+          .then((resp) => {})
+          .catch((pageError) => console.log('Error setting page', pageError));
         })
-        .catch((error) => console.log('Error starting Google Analytics', error));
+        .catch((error) => console.log(`setCurrentPage: ${this.analyticsStartupError}`, error));
     });
   }
 
-  logEvent(category: string, event: string, params?: any) {
+  logEvent(category: string, event: string, label?: string, params?: any) {
     this.platform.ready().then(() => {
       this.ga
         .startTrackerWithId(this.googleAnalyticsKey)
         .then(() => {
-          this.ga.trackEvent(category, event).then((resp) => {});
+          this.ga.trackEvent(category, event, label)
+          .then((resp) => {})
+          .catch((eventError) => console.log('Error tracking event', eventError));
         })
-        .catch((error) => console.log('Error starting Google Analytics', error));
+        .catch((error) => console.log(`logEvent: ${this.analyticsStartupError}`, error));
     });
   }
 
-  logError(message: string) {}
+  addCustomDimension(key: number, value: string) {
+    this.ga
+    .startTrackerWithId(this.googleAnalyticsKey)
+    .then(() => {
+      this.ga.addCustomDimension(key,value)
+        .then((resp) => {})
+        .catch((dimError) => console.log('Error adding custom dimension ', dimError));
+    })
+    .catch((error) => console.log(`addCustomDimension: ${this.analyticsStartupError}`, error));
+  }
+
+  logError(type: string, message: string) {
+    this.logEvent(AnalyticsEventCategories.ERROR, type, message);
+  }
 
-  setUserId(userId: string) {}
+  logException(message: string, fatal: boolean) {
+    this.ga
+    .startTrackerWithId(this.googleAnalyticsKey)
+    .then(() => {
+      this.ga.trackException(message, fatal)
+        .then((resp) => {})
+        .catch((trackingError) => console.log('Error logging exception in Google Analytics', trackingError));
+    })
+    .catch((error) => console.log(`logException: ${this.analyticsStartupError}`, error));
+  }
+
+  setUserId(userId: string) {
+    this.ga
+    .startTrackerWithId(this.googleAnalyticsKey)
+    .then(() => {
+      this.ga.setUserId(userId)
+        .then((resp) => {})
+        .catch((idError) => console.log(`Error setting userid ${userId}`,idError));
+    })
+    .catch((error) => console.log(`setUserId: ${this.analyticsStartupError}`, error));
+  }
+
+  getDiffDays(userDate: string): number {
+    const today = moment().startOf('day');
```

Comment:
We recently introduced the `DateTime` wrapper to constrain our usage of `moment` to a single place. Could you use this class please? You might need to extend it with duration functionality.

Reply:
modified to use DateTime class.